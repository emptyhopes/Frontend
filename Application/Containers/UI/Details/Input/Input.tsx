import * as React from "react";

import styled from "styled-components";

import { PropsInterface } from "@/Application/Containers/UI/Details/Input/Types/PropsInterface";

const StyledInput = styled.input`
  width: 500px;
  padding: 14px;
  border: 1px solid #c0c0c0;
  border-radius: 12px;
  color: #000000;
  outline: none;

  &:focus {
    border: 1px solid #00bfff;
  }
`;

const DefaultInput = (props: PropsInterface, ref) => {
  return (
    <>
      <StyledInput ref={ref} {...props} />
    </>
  );
};

const Input = React.forwardRef(DefaultInput);

export { Input };
