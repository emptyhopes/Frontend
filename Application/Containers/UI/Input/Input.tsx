import styled from "styled-components";

import { FunctionComponent } from "react";

import { InputInterface } from "@/Application/Containers/UI/Input/Types/InputInterface";

const StyledInput = styled.input``;

const Input: FunctionComponent<InputInterface> = (props) => {
  return <StyledInput {...props} />;
};

export { Input };
