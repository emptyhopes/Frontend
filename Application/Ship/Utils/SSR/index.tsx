import * as React from "react";
import * as ReactDOM from "react-dom";

import { Application } from "@/Application/Containers/Application/Application";

const Structures = <Application />;

const element = document.getElementById("root");

ReactDOM.hydrate(Structures, element);
