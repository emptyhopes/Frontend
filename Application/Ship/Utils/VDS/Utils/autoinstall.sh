#! /bin/bash

# Update and upgrade system
sudo apt-get update --yes && sudo apt-get upgrade --yes

# Remove not use packages
sudo apt-get autoremove --yes

# Install packages
sudo apt-get install git nginx certbot python3-certbot-nginx --yes

# SWAP

sudo fallocate -l 2048M /swapfile
sudo chmod 600 /swapfile
sudo mkswap /swapfile
sudo swapon /swapfile

sudo cp /etc/fstab /etc/fstab.back

echo "/swapfile none swap sw 0 0" | sudo tee -a /etc/fstab

# NVM ( node@latest and npm@latest )

curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.2/install.sh | bash

source ~/.nvm/nvm.sh

nvm install node

npm install --global npm@latest

# Install letsencrypt SSL certificate

sudo certbot
