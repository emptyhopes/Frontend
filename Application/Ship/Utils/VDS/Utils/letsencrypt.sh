#! /bin/bash

hostname="emptyhopes.ru"

sudo certbot certonly --nginx -d $hostname