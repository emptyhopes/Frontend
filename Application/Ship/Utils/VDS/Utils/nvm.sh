#! /bin/bash

temporary=$(mktemp --directory)

curl --silent --location https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.2/install.sh >"$temporary/install.sh"

bash "$temporary/install.sh"

source ~/.nvm/nvm.sh

nvm install node
