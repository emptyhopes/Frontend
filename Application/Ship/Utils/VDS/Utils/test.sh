#! /bin/bash

hostname="gartmann-art.ru"

curl --location -I http://$hostname.ru
curl --location -I http://en.$hostname.ru
curl --location -I http://ru.$hostname.ru

curl --location -I http://www.$hostname.ru
curl --location -I http://www.en.$hostname.ru
curl --location -I http://www.ru.$hostname.ru

curl --location -I https://$hostname.ru
curl --location -I https://en.$hostname.ru
curl --location -I https://ru.$hostname.ru

curl --location -I https://www.$hostname.ru
curl --location -I https://www.en.$hostname.ru
curl --location -I https://www.ru.$hostname.ru