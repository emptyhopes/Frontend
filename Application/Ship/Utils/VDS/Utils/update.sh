#! /bin/bash

sudo apt-get update --yes && sudo apt-get upgrade --yes
sudo apt-get autoremove --yes

sudo apt-get install git nginx certbot python3-certbot-nginx --yes