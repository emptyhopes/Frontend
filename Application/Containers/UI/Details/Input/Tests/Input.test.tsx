import { RenderTests } from "@/Application/Ship/Tests/Utils/RenderTests";

import { Input } from "@/Application/Containers/UI/Details/Input/Input";

test("Input", () => {
  const { container } = RenderTests(<Input type={""} placeholder={""} />);

  expect(container).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
