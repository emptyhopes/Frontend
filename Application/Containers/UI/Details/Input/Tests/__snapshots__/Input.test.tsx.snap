// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input 1`] = `
.c0 {
  width: 500px;
  padding: 14px;
  border: 1px solid #c0c0c0;
  border-radius: 12px;
  color: #000000;
  outline: none;
}

.c0:focus {
  border: 1px solid #00bfff;
}

<div>
  <input
    class="c0"
    placeholder=""
    type=""
  />
</div>
`;
